uvicorn/main.py

```diff
@@ -37,8 +38,47 @@
 )
 
 
+def _setup_defaults_from_config_script(
+    ctx: click.Context, param: click.Parameter, value: typing.Optional[str]
+):
+    if value is None:
+        return
+
+    members: dict = runpy.run_path(value)
+
+    try:
+        config: Config = members["config"]
+    except KeyError:
+        raise click.BadParameter(f"no 'config' object declared in {value}")
+
+    if not isinstance(config, Config):
+        raise click.BadParameter(
+            f"'config' must be a 'uvicorn.Config' object (got {type(config)})"
+        )
+
+    # Read parameter defaults from the config object.
+    # Values may still be overridden using command line arguments.
```

Comment:
How's this instead, for loading names at the top-level?...

```python
members: dict = runpy.run_path(value)

valid_param_names = inspect.signature(main).parameters.keys()
ctx.default_map = {
    param.name: getattr(config, param.name)
    for param in ctx.command.params
    if hasattr(param.name in valid_param_names)
}
```

Reply:
I guess that maybe doesn't work right? Given that we still want type conversion to be applied?